pkg/collector/corechecks/gpu/nvmlmetrics/collector.go

```diff
@@ -88,8 +75,10 @@ func newCollectorWithSubsystems(lib nvml.Interface, subsystems map[string]subsys
 			return nil, fmt.Errorf("failed to get device handle for index %d: %s", i, nvml.ErrorString(ret))
 		}
 
+		tags := getTagsFromDevice(dev)
+
 		for name, factory := range subsystems {
```

Comment:
rename the factory to builder or creator as it more suits the purpose of the value type of this map.
the factory in this case is the whole map (`allSubsystems`)

Reply:
Renamed, also renamed the `subsystemFactory` type to match.